samples/dotnet_core_managed_gateway/Program.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.IO;
+using Microsoft.Azure.Devices.Gateway;
+
+namespace dotnet_core_managed_gateway
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            try
+            {
+                if (args.Length < 1 || File.Exists(args[0]) == false)
+                {
+                    Console.Error.WriteLine("Usage: managed-gateway <<path to json>>");
+                }
+                else
+                {
+                    var json_path = args[0];
+                    var gateway = GatewayInterop.CreateFromJson(json_path);
+
+                    Console.WriteLine(".NET Core Gateway is running. Press return to quit.");
+                    Console.ReadLine();
+
+                    GatewayInterop.Destroy(gateway);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine($"ERROR: {ex.ToString()}");
```

Comment:
Since we aren't doing anything interesting in the `catch` block, the try/catch stuff is just a distraction. I'd remove it from the sample. If the sample throws, the program will exit with the same information that your `Console.Error.Writeline` statement spits out (or better).